### PR TITLE
SONARPY-2193: Update sonarlint-core.version to v10 (major)

### PR DESCRIPTION
**Pull request by joke1196**

### CODE REVIEW

File: its/plugin/it-python-plugin-test/src/test/java/com/sonar/python/it/plugin/SonarLintTest.java [hunk 1]
```diff
@@ -59,50 +67,64 @@ class SonarLintTest {
...
+        new ActiveRule("python:BackticksUsage", SonarLanguage.PYTHON.name()))
+      .build();
+
+    Map<Level, List<String>> logsByLevel = new HashMap<>();
+    LogOutput logOutput = new LogOutput() {
+      @Override
+      public void log(String formattedMessage, Level level, @Nullable String stacktrace) {
+        List<String> logs = logsByLevel.getOrDefault(level, new ArrayList<>());
+        logs.add(formattedMessage);
+        logsByLevel.putIfAbsent(level, logs);
```
Comment (Seppli11) [inline its/plugin/it-python-plugin-test/src/test/java/com/sonar/python/it/plugin/SonarLintTest.java:126]: I find the version above a bit more concise
```suggestion
        logsByLevel.computeIfAbsent(level, k -> new ArrayList<>()).add(formattedMessage);
```

File: its/plugin/it-python-plugin-test/src/test/java/com/sonar/python/it/plugin/IPythonTest.java
Comment (Seppli11) [inline its/plugin/it-python-plugin-test/src/test/java/com/sonar/python/it/plugin/IPythonTest.java:1]: Would it maybe make sense to change this class's name to `SonarLintIPythonTest` or similar, to reflect that it does the something as `SonarLint` but for ipython?

**Review by Seppli11 — APPROVED** [review]
I left two comments about some small stuff.
Apart from that, LGTM!

File: its/plugin/it-python-plugin-test/src/test/java/com/sonar/python/it/plugin/IPythonTest.java
Comment (joke1196) [reply its/plugin/it-python-plugin-test/src/test/java/com/sonar/python/it/plugin/IPythonTest.java:1]: Yes I think it makes a lot of sense!

**Pull request merged**